handler_error.go

```diff
@@ -17,6 +17,7 @@ const (
 	defaultTemplateName   = "error"
 	defaultTemplateFormat = "json"
 	defaultContentType    = "application/json"
+	defaultVersionHeader  = "X-Tyk-Default-Version"
```

Comment:
Let's make it configurable, and add this header ONLY if header value is specified. 

Related to our latest tech meeting. Basic idea that we should not expose user infrastructure, in other words that user use Tyk (which can be easily identified by this header name).

Reply:
So should we have options for header, url-param and to disable?

Reply:
Doing it like this for now so relevant part of apidef would be configurable as:


"version_data": {
    "not_versioned": false,
    "default_version": "v2",
    "versions": {
        "v1": {"name": "v1", ...},
        "v2": {"name": "v2", ...}
    }
    "DefaultLocation": "url-param"
    "DefaultKey": "dv"
}

Reply:
Got it, that makes much more sense than what i was doing, cheers.